app/assets/js/char-count.js

```diff
@@ -251,23 +251,26 @@
     }
 
     // Update message
-
-    if (options && options.language) {
-        // placeholder - pending welsh translation - will be a string of either "en" or "cy"
-    }
-
-    var charVerb = 'remaining'
-    var charNoun = 'character'
-    var displayNumber = remainingNumber
-    if (options && options.wordCount) {
-      charNoun = 'word'
+    var prefixClause, countVerb, countNoun
+    var displayNumber = Math.abs(remainingNumber) // postive count of numbers
+    if (options && options.language && options.language==='cy') {      // if set, will be a string of either "en" or "cy"
+        prefixClause = "Mae gennych "
+        countVerb = (remainingNumber < 0) ? 'yn ormod' : 'yn weddill'
+        if (options && options.wordCount) {
+            countNoun = ((remainingNumber === -1 || remainingNumber === 1) ? 'gair' : 'geiriau')
+        } else {
+            countNoun = ((remainingNumber === -1 || remainingNumber === 1) ? 'cymeriad' : 'o gymeriadau')
+        }
+    } else {
+        prefixClause = "You have "
+        countVerb = (remainingNumber < 0) ? 'too many' : 'remaining'
+        if (options && options.wordCount) {
+            countNoun = ((remainingNumber === -1 || remainingNumber === 1) ? 'word' : 'words')
+        } else {
+            countNoun = ((remainingNumber === -1 || remainingNumber === 1) ? 'character' : 'characters')
+        }
```

Comment:
The logic to form the whole message is repeated, can we try to combine the messages using inline if-else? We can check which one looks betters and chose the appropriate solution.